src/Model/Introduction.php

```diff
@@ -79,43 +76,40 @@ public function confirm(bool $duplex = false, bool $hidden = null)
 			'protocol'  => $protocol,
 			'writable'  => $writable,
 			'hidden'    => $hidden ?? $contact['hidden'],
-			'rel'       => $new_relation,
+			'rel'       => $newRelation,
 		];
 		$this->dba->update('contact', $fields, ['id' => $contact['id']]);
 
 		array_merge($contact, $fields);
 
-		if ($new_relation == Contact::FRIEND) {
+		if ($newRelation == Model\Contact::FRIEND) {
 			if ($protocol == Protocol::DIASPORA) {
-				$ret = Diaspora::sendShare(User::getById($contact['uid']), $contact);
+				$ret = Diaspora::sendShare(Model\Contact::getById($contact['uid']), $contact);
 				$this->logger->info('share returns', ['return' => $ret]);
 			} elseif ($protocol == Protocol::ACTIVITYPUB) {
 				ActivityPub\Transmitter::sendActivity('Follow', $contact['url'], $contact['uid']);
 			}
 		}
 
-		$this->delete();
+		return true;
```

Comment:
If we inject the `Introduction` Repository in it's model, we could make it like:
```suggestion
		return $this->repo->delete($this);
```

Reply:
Ah, good, I was wondering how to handle this and keep a one-liner in the Module.

Reply:
This doesn't blank out the model though. The `Repository::delete` uses an argument by reference but passing `$this` doesn't nullify the Model object as expected.

Reply:
That's true.. Hmmm... Tbh I don't have an idea for now except making this in the module:
```php
If ($intro->delete()) {
    unset($intro);
}
```